sky/backends/backend_utils.py

```diff
@@ -1770,6 +1770,7 @@ def check_owner_identity(cluster_name: str) -> None:
                         'allow the operation as the two identities are likely to have '
                         'the same access to the cluster, but please be aware that '
                         'this is not guaranteed.')
+                if i != 0 or len(owner_identity) != len(current_user_identity):
```

Comment:
Add a comment on why these conditions?

also “this is not guaranteed “ seems vague; rephrase possible?

Reply:
Good point! Added. PTAL. : )